src/tools/tool.ts

```diff
@@ -133,6 +179,28 @@ export class Tool<Args extends ZodRawShape | undefined = undefined> {
 }
 
 function getErrorResult(requestId: RequestId, error: unknown): CallToolResult {
+  if (error instanceof ZodiosError && isZodErrorLike(error.cause)) {
+    // Schema validation errors on otherwise successful API calls will not return an "error" result to the MCP client.
+    // We instead return the full response from the API with a data quality warning message
+    // that mentions why the schema validation failed.
+    // This should make it so users don't get "stuck" when our schemas are too strict or wrong.
+    // The only con is that the full response from the API might be larger than normal
+    // since a successful schema validation "trims" the response down to the shape of the schema.
+    const validationError = fromError(error.cause);
```

Comment:
`zod-validation-error` also automatically limits the number of issues included in the error message. For example, say the API returned an array with 100 object, where each object had some validation error. If we returned the Zod issue message to the client, it would be enormous and exhaustively return a message for all 100 objects. With `zod-validation-error`, it defaults to truncating to 10 objects.